crates/astria-composer/src/composer.rs

```diff
@@ -93,17 +109,24 @@ impl Composer {
     /// See `[from_config]` for its error scenarios.
     pub async fn from_config(cfg: &Config) -> eyre::Result<Self> {
         let (composer_status_sender, _) = watch::channel(Status::default());
+        let shutdown_token = CancellationToken::new();
+
         let (executor, executor_handle) = Executor::new(
             &cfg.sequencer_url,
             &cfg.private_key,
             cfg.block_time_ms,
             cfg.max_bytes_per_bundle,
+            shutdown_token.clone(),
         )
         .wrap_err("executor construction from config failed")?;
 
-        let grpc_server = GrpcServer::new(cfg.grpc_addr, executor_handle.clone())
-            .await
-            .wrap_err("failed to create grpc server")?;
+        let grpc_server = GrpcServer::new(
+            cfg.grpc_addr,
+            executor_handle.clone(),
+            shutdown_token.clone(),
```

Comment:
I think this shutdown token on the gRPC server is too simple. It fulfills a similar role as the API server, so it should do two things:

1. stop receiving requests on its composer service,
2. but keep serving requests on its health service.

Reply:
hmm i had thought of this initially but I thought it was fine just letting the gRPC server shutdown gracefully as it ll take these services down. let me try doing what you suggested.